app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/DefaultTasksRepository.kt

```diff
@@ -131,11 +115,9 @@ class DefaultTasksRepository(
     }
 
     override suspend fun deleteAllTasks() {
```

Comment:
If the function is already marked `suspend`, I wonder if we still need the `coroutineScope`, and to explicitly `launch` again in it.

Reply:
Yeah, I'm not sure why this was implemented using `coroutineScope`. Presumably so that if one job fails then the other one is immediately cancelled, but that's not really what we want. 

I will add offline-first lazy writes in a separate PR which will resolve this issue.

Reply:
It was implemented using `coroutineScope` so that you could `launch` multiple coroutines in parallel

Reply:
Gotcha, thanks.